Frontend Index.js Refactor

ehsonp opened the pull request:
Edward, Waynar, and I worked on refactoring the index.js file to be much simpler. The router now contains the url and pageType variables, along with the router calling the different setup state functions from the state folder instead of importing it from index. 

Additionally I have a very small frame setup for Ethan and I's log component we are building.

forschoolandnothingelse committed:
backend: changes backend to use session cookies

The previous implementation of the backend saved passwords in plaintext
on the clientside. This was really bad for a number of reasons, not
least of which is it just didn't work on a ton of browsers.

The new backend uses an /auth route to first authenticate a user. It
expects a JSON object of the form:
{
    email: user@email.com,
    pwd: qwerty
}
From here it either authenticates the user's session and returns a JSON
object with the "error" field set to null, or does not authenticate the
user's session and returns an object with the "error" field set to an
error string.

This also changes the way the /{create,read,update,delete}User routes
works. Now all user routes are just under /user. The way we do CRUD for
users is using HTTP methods, POST for create, GET for read, PUT for
update, and DELETE for delete. This should enforce not making every API
call a POST for no good reason.

This also enforces better security since a user can only modify their
own user entry as determined for the session rather than the user
themselves.

The way user encryption works changes as well. Previously because the
user always sent their password, the password was used as the key. Now a
special encryption key is saved. The key is a simple encrypt-then-MAC of
the user's password and the sessionSecret. This will break all
previously stored data in Bujo, but makes for better security in the
long run.

Of course, we need to add an envvar for the sessionSecret under the key
SESSION_SECRET, or deployment to production will fail.

forschoolandnothingelse committed:
backend: changes backend to use session cookies

The previous implementation of the backend saved passwords in plaintext
on the clientside. This was really bad for a number of reasons, not
least of which is it just didn't work on a ton of browsers.

The new backend uses an /auth route to first authenticate a user. It
expects a JSON object of the form:
{
    email: user@email.com,
    pwd: qwerty
}
From here it either authenticates the user's session and returns a JSON
object with the "error" field set to null, or does not authenticate the
user's session and returns an object with the "error" field set to an
error string.

This also changes the way the /{create,read,update,delete}User routes
works. Now all user routes are just under /user. The way we do CRUD for
users is using HTTP methods, POST for create, GET for read, PUT for
update, and DELETE for delete. This should enforce not making every API
call a POST for no good reason.

This also enforces better security since a user can only modify their
own user entry as determined for the session rather than the user
themselves.

The way user encryption works changes as well. Previously because the
user always sent their password, the password was used as the key. Now a
special encryption key is saved. The key is a simple encrypt-then-MAC of
the user's password and the sessionSecret. This will break all
previously stored data in Bujo, but makes for better security in the
long run.

Of course, we need to add an envvar for the sessionSecret under the key
SESSION_SECRET, or deployment to production will fail.

forschoolandnothingelse committed:
backend: changes backend to use session cookies

The previous implementation of the backend saved passwords in plaintext
on the clientside. This was really bad for a number of reasons, not
least of which is it just didn't work on a ton of browsers.

The new backend uses an /auth route to first authenticate a user. It
expects a JSON object of the form:
{
    email: user@email.com,
    pwd: qwerty
}
From here it either authenticates the user's session and returns a JSON
object with the "error" field set to null, or does not authenticate the
user's session and returns an object with the "error" field set to an
error string.

This also changes the way the /{create,read,update,delete}User routes
works. Now all user routes are just under /user. The way we do CRUD for
users is using HTTP methods, POST for create, GET for read, PUT for
update, and DELETE for delete. This should enforce not making every API
call a POST for no good reason.

This also enforces better security since a user can only modify their
own user entry as determined for the session rather than the user
themselves.

The way user encryption works changes as well. Previously because the
user always sent their password, the password was used as the key. Now a
special encryption key is saved. The key is a simple encrypt-then-MAC of
the user's password and the sessionSecret. This will break all
previously stored data in Bujo, but makes for better security in the
long run.

Of course, we need to add an envvar for the sessionSecret under the key
SESSION_SECRET, or deployment to production will fail.

forschoolandnothingelse committed:
backend: changes backend to use session cookies

The previous implementation of the backend saved passwords in plaintext
on the clientside. This was really bad for a number of reasons, not
least of which is it just didn't work on a ton of browsers.

The new backend uses an /auth route to first authenticate a user. It
expects a JSON object of the form:
{
    email: user@email.com,
    pwd: qwerty
}
From here it either authenticates the user's session and returns a JSON
object with the "error" field set to null, or does not authenticate the
user's session and returns an object with the "error" field set to an
error string.

This also changes the way the /{create,read,update,delete}User routes
works. Now all user routes are just under /user. The way we do CRUD for
users is using HTTP methods, POST for create, GET for read, PUT for
update, and DELETE for delete. This should enforce not making every API
call a POST for no good reason.

This also enforces better security since a user can only modify their
own user entry as determined for the session rather than the user
themselves.

The way user encryption works changes as well. Previously because the
user always sent their password, the password was used as the key. Now a
special encryption key is saved. The key is a simple encrypt-then-MAC of
the user's password and the sessionSecret. This will break all
previously stored data in Bujo, but makes for better security in the
long run.

Of course, we need to add an envvar for the sessionSecret under the key
SESSION_SECRET, or deployment to production will fail.

forschoolandnothingelse committed:
backend: changes backend to use session cookies

The previous implementation of the backend saved passwords in plaintext
on the clientside. This was really bad for a number of reasons, not
least of which is it just didn't work on a ton of browsers.

The new backend uses an /auth route to first authenticate a user. It
expects a JSON object of the form:
{
    email: user@email.com,
    pwd: qwerty
}
From here it either authenticates the user's session and returns a JSON
object with the "error" field set to null, or does not authenticate the
user's session and returns an object with the "error" field set to an
error string.

This also changes the way the /{create,read,update,delete}User routes
works. Now all user routes are just under /user. The way we do CRUD for
users is using HTTP methods, POST for create, GET for read, PUT for
update, and DELETE for delete. This should enforce not making every API
call a POST for no good reason.

This also enforces better security since a user can only modify their
own user entry as determined for the session rather than the user
themselves.

The way user encryption works changes as well. Previously because the
user always sent their password, the password was used as the key. Now a
special encryption key is saved. The key is a simple encrypt-then-MAC of
the user's password and the sessionSecret. This will break all
previously stored data in Bujo, but makes for better security in the
long run.

Of course, we need to add an envvar for the sessionSecret under the key
SESSION_SECRET, or deployment to production will fail.

forschoolandnothingelse committed:
backend: changes backend to use session cookies

The previous implementation of the backend saved passwords in plaintext
on the clientside. This was really bad for a number of reasons, not
least of which is it just didn't work on a ton of browsers.

The new backend uses an /auth route to first authenticate a user. It
expects a JSON object of the form:
{
    email: user@email.com,
    pwd: qwerty
}
From here it either authenticates the user's session and returns a JSON
object with the "error" field set to null, or does not authenticate the
user's session and returns an object with the "error" field set to an
error string.

This also changes the way the /{create,read,update,delete}User routes
works. Now all user routes are just under /user. The way we do CRUD for
users is using HTTP methods, POST for create, GET for read, PUT for
update, and DELETE for delete. This should enforce not making every API
call a POST for no good reason.

This also enforces better security since a user can only modify their
own user entry as determined for the session rather than the user
themselves.

The way user encryption works changes as well. Previously because the
user always sent their password, the password was used as the key. Now a
special encryption key is saved. The key is a simple encrypt-then-MAC of
the user's password and the sessionSecret. This will break all
previously stored data in Bujo, but makes for better security in the
long run.

Of course, we need to add an envvar for the sessionSecret under the key
SESSION_SECRET, or deployment to production will fail.

forschoolandnothingelse committed:
backend: changes backend to use session cookies

The previous implementation of the backend saved passwords in plaintext
on the clientside. This was really bad for a number of reasons, not
least of which is it just didn't work on a ton of browsers.

The new backend uses an /auth route to first authenticate a user. It
expects a JSON object of the form:
{
    email: user@email.com,
    pwd: qwerty
}
From here it either authenticates the user's session and returns a JSON
object with the "error" field set to null, or does not authenticate the
user's session and returns an object with the "error" field set to an
error string.

This also changes the way the /{create,read,update,delete}User routes
works. Now all user routes are just under /user. The way we do CRUD for
users is using HTTP methods, POST for create, GET for read, PUT for
update, and DELETE for delete. This should enforce not making every API
call a POST for no good reason.

This also enforces better security since a user can only modify their
own user entry as determined for the session rather than the user
themselves.

The way user encryption works changes as well. Previously because the
user always sent their password, the password was used as the key. Now a
special encryption key is saved. The key is a simple encrypt-then-MAC of
the user's password and the sessionSecret. This will break all
previously stored data in Bujo, but makes for better security in the
long run.

Of course, we need to add an envvar for the sessionSecret under the key
SESSION_SECRET, or deployment to production will fail.

forschoolandnothingelse committed:
backend: changes backend to use session cookies

The previous implementation of the backend saved passwords in plaintext
on the clientside. This was really bad for a number of reasons, not
least of which is it just didn't work on a ton of browsers.

The new backend uses an /auth route to first authenticate a user. It
expects a JSON object of the form:
{
    email: user@email.com,
    pwd: qwerty
}
From here it either authenticates the user's session and returns a JSON
object with the "error" field set to null, or does not authenticate the
user's session and returns an object with the "error" field set to an
error string.

This also changes the way the /{create,read,update,delete}User routes
works. Now all user routes are just under /user. The way we do CRUD for
users is using HTTP methods, POST for create, GET for read, PUT for
update, and DELETE for delete. This should enforce not making every API
call a POST for no good reason.

This also enforces better security since a user can only modify their
own user entry as determined for the session rather than the user
themselves.

The way user encryption works changes as well. Previously because the
user always sent their password, the password was used as the key. Now a
special encryption key is saved. The key is a simple encrypt-then-MAC of
the user's password and the sessionSecret. This will break all
previously stored data in Bujo, but makes for better security in the
long run.

Of course, we need to add an envvar for the sessionSecret under the key
SESSION_SECRET, or deployment to production will fail.

ehsonp committed:
meetings: added all of our meeting notes to github

ehsonp committed:
frontend: many refactoring changes to router and index

ehsonp committed:
frontend: refactor index.js to have its functions placed in state folder

ehsonp committed:
frontend: refactoring linting error fix

ehsonp commented:
Do not merge ran into some bugs already with this refactor give me a day and ill fix these bugs

ehsonp committed:
front-end/router: bug with refactoring imported contentwrapper incorrectly

forschoolandnothingelse commented on src/front-end/components/log.jsx:
```diff
@@ -0,0 +1,45 @@
+/* eslint-enable */
+
+let template = <template>
+	<link type="text/css" rel="stylesheet" href="log.css" />
```
You should remove this line since the client will waste time making a network request to get this file, which is just empty.
You should also delete `src/front-end/components/log.css` so nobody tries to use it.

waynarbocangel replied:
This is an incomplete file that he is working on for the next commit. I would ignore it for now since he will be putting stuff in them over the next couple of days but yeah thanks for pointing it out!

forschoolandnothingelse replied:
I think it's worth it to do it for now, since it's always possible things will change and this may end up forgotten about.

ehsonp replied:
honestly i barely wrote code for it so its ok to remove, i agree it professionally shouldnt be there

forschoolandnothingelse commented on src/back-end/schema.js:
```diff
@@ -76,10 +76,10 @@ const userSchema = {
 			objectType: String,
 			parent: String,
 			date: Date,
+			content: [String],
```
I just want to make a note that this change means we need to drop the backend again.

ehsonp replied:
Let me discuss this with waynar, i might remove it since it makes more sense to implement in your changes rather than mine. But we agreed this would be a better way to store it

waynarbocangel replied:
Yeah I'm taking charge of this refactor. Delete it from here please Ehson, thanks!

ehsonp replied:
Will do 
and ill fix linting again

forschoolandnothingelse reviewed (changes requested):
Just make that one change and make sure this passes linting. I'll test the PR and then get back with an approval once I see it works.

ehsonp committed:
backend/schema.js: fixed small change and lint issue

ehsonp committed:
frontend/log: removed the component

forschoolandnothingelse commented:
I'm going to approve this, but for future use `git rebase <branch>` instead of `git merge <branch>` so we don't have to deal with merge commits.

waynarbocangel reviewed (approved):
Great work Ehson! Looks like everything split up really nicely.

ehsonp committed:
frontend/state: rebase errors force pushing current state

ehsonp committed:
frontend: rebase docs fix

ehsonp commented:
Refactored a lot of code in frontend, changed how router and page are updated. State updates display in frontend/state folder. plethora of rebasing issues
